src/rebuild-parse-cache/src/get-repos.ts

```diff
@@ -0,0 +1,173 @@
+    '/atlas/username',
+    '/atlas/host',
+    '/atlas/dbname',
+    '/github/bot/password',
+  ];
+
+  const pathToEnvMap: Map<string, string> = new Map();
+
+  pathToEnvMap.set(`${ssmPrefix}/atlas/password`, 'MONGO_ATLAS_PASSWORD');
+  pathToEnvMap.set(`${ssmPrefix}/atlas/username`, 'MONGO_ATLAS_USERNAME');
+  pathToEnvMap.set(`${ssmPrefix}/atlas/host`, 'MONGO_ATLAS_HOST');
+  pathToEnvMap.set(`${ssmPrefix}/atlas/dbname`, 'MONGO_ATLAS_DBNAME');
+  pathToEnvMap.set(`${ssmPrefix}/github/bot/password`, 'GITHUB_BOT_PASSWORD');
+
+  const parametersMap: Record<string, string> = {};
+
+  await Promise.all(
+    parameters.map(async paramName => {
+      const fullParamPath = `${ssmPrefix}${paramName}`;
+      const getParamCommand = new GetParameterCommand({
+        Name: fullParamPath,
+        WithDecryption: true,
+      });
+
+      const envName = pathToEnvMap.get(fullParamPath);
+
+      try {
+        const ssmResponse = await ssmClient.send(getParamCommand);
+        const paramString = ssmResponse.Parameter?.Value;
+
+        if (!envName || !paramString) {
+          console.error(
+            `ERROR! Could not retrieve string for the following param: ${paramName}`,
+          );
+          return;
+        }
+
+        core.setSecret(paramString);
+        parametersMap[envName] = paramString;
+      } catch (e) {
+        console.error(`Could not retrieve ${envName}`);
+      }
+    }),
+  );
+  return parametersMap;
+}
+
+async function getMongoClient({
+  MONGO_ATLAS_PASSWORD,
+  MONGO_ATLAS_USERNAME,
+  MONGO_ATLAS_HOST,
+}: Record<string, string>): Promise<MongoClient> {
+  const atlasUrl = `mongodb+srv://${MONGO_ATLAS_USERNAME}:${MONGO_ATLAS_PASSWORD}@${MONGO_ATLAS_HOST}/admin?retryWrites=true`;
+  core.setSecret(atlasUrl);
+  const client = new MongoClient(atlasUrl);
+  try {
+    const connectedClient = await client.connect();
+
+    process.on('SIGTERM', async () => {
+      await connectedClient.close();
+    });
+
+    return connectedClient;
+  } catch (e) {
+    throw new Error('Failed to connect to DB');
+  }
+}
+
+export async function getApiKey(): Promise<string> {
+  const API_KEY_NAME = 'zxccutwa64';
+
+  const client = new APIGatewayClient({ region: REGION });
+  const command = new GetApiKeyCommand({
+    apiKey: API_KEY_NAME,
+    includeValue: true,
+  });
+
+  const { value } = await client.send(command);
+  if (!value) throw new Error('No value found for API key');
+
+  core.setSecret(value);
+  return value;
+}
+
+export interface RepoInfo {
+  repoOwner: string;
+  repoName: string;
+}
+interface RepoBranchEntry {
+  repoName: string;
+}
+
+interface FindRepoResponse {
+  search: {
+    repos: Array<{
+      repo: { owner: { login: string } };
+    }>;
+  };
+}
+/**
+ * Get repos branches entries from atlas and verify owner by
+ * using the GitHub API to search for the repo URL.
+ */
+export async function getRepos(): Promise<RepoInfo[]> {
+  const parameters = await getParameters('dotcomstg');
```

Comment:
Do we always want this to be `dotcomstg`?

Reply:
This is a good question. This will only be used for a single environment, which I believe should be `dotcomstg`. The cache is only used for staging builds on the `master`/`main` branch of any given doc site.